### DISCUSSION TITLE
Commercial License + combined mutations

### DISCUSSION THREAD

**Discussion by jadolfbr** [brianhie/efficient-evolution#4] [Q&A]
Hello,

The CC4.0 is a very restrictive licence for a project.  I was wondering if you are sticking with that license (even though most of the code is a wrapper to using ESM-1b/v) or if you have any simple commercial licenses available through your tech transfer office.  

I was also wondering why the subsequent mutations in your paper were manually put together.  It seems like it would be possible to do a combinatorial scoring and order best-scoring hits from the manifold.  Did I miss something there?

Thanks,
-Jared

**Comment by brianhie** [comment]
Thanks for reaching out @jadolfbr, and apologies for the late response!

I'll keep you updated about the license, and I'll certainly look into alternatives.

For the round 2 combinations, in the MEDI (UCA) trajectories we did just test all possible combinations. For the mAb114 and SARS-CoV-2 trajectories, combinations were chosen to get better diversity of edit distances than would be suggested by just, e.g., computing pseudolikelihoods, which [tends to prefer just adding all of the mutations together](https://www.nature.com/articles/s41587-021-01146-5), though this is indeed possible to do. Manually combining single-site mutations based on a first round is common in traditional engineering campaigns and was a natural choice to explore a more diverse subset of the combinatorial space in a single round given our very low-throughput pipeline. With higher throughput, I do think proceeding directly to a combinatorial library of the language model-proposed mutations could be quite effective.
